### ISSUE TITLE
Composite Keys not working

### ISSUE THREAD

**Issue by chris-eaton** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#231]
Using the Reverse POCO generator with Composite Keys does not work.  The generated code is

```
[Required]
[Key]
[Display(Name = "Driver ID")]
public int DriverId { get; set; } // DriverId (Primary key)

[Required]
[Key]
[Display(Name = "Product group ID")]
public int ProductGroupId { get; set; } // ProductGroupId (Primary key)
```

But these two fields are actually a Composite Key and running the app then gives an error "Unable to determine composite primary key ordering"

I would expect it should generate code more like [I removed other properties for brevity]
```
[Key, Column(Order = 1)]
public int DriverId { get; set; } // DriverId (Primary key)

[Key, Column(Order = 2)]
public int ProductGroupId { get; set; } // ProductGroupId (Primary key)
```

Is there anything we can do to mitigate this?

**Comment by sjh37** [comment]
For now, turn off data annotations.

I'll get the column order added to the Key data annotation.

**Comment by sjh37** [comment]
You need to turn on the following
UseDataAnnotationsSchema = false;
That causes the Column order to be generated.

**Comment by sjh37** [comment]
I've updated the code to solve your problem.
Pull down the changes above into your code

**Comment by chris-eaton** [comment]
Thank you for the fast response!

**Comment by cody1024d** [comment]
Sorry for commenting on a closed issue; but I am curious when this will be released?

**Comment by sjh37** [comment]
I usually wait until I've got a few enhancements in. I'll try and get this released next week.

**Comment by sjh37** [comment]
Released

**Comment by PedroMoraisCosta** [comment]
Hi is this already released ? 
I have updated my poco in VS but the *.tt file stills not create the Column Order attribute... Am I doing something wrong ?

**Comment by ErikEJ** [comment]
Did you update all three files?

**Comment by sjh37** [comment]
This was fixed in https://github.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/releases/tag/v2.29.1

Can you check your version numbers on all 3 tt/ttinclude files please and let me know what they are

**Comment by PedroMoraisCosta** [comment]
Hi. I have found my problem. 
Thanks for the help.

**Comment by nick5454** [comment]
> Hi. I have found my problem.
> Thanks for the help.

Do you remember what you did? I'm having the same issue

**Comment by PedroMoraisCosta** [comment]
Sorry but that was too long ago in 2017.. I have no idea what was the
solution...



Nick ***@***.***> escreveu no dia quinta, 13/05/2021 à(s)
14:28:

**Comment by sjh37** [comment]
@PedroMoraisCosta what version are you having problems with?

**Comment by nick5454** [comment]
@sjh37 With the latest version, it is not putting the Column Order attribute on the column which causes an issue.

**Comment by sjh37** [comment]
@nick5454 I will fix this..

**Comment by sjh37** [comment]
@nick5454 Done. 
Modify your `EF.Reverse.POCO.v3.ttinclude` file and change:
```cs
if (Settings.UseDataAnnotations)
{
    Settings.AdditionalNamespaces.Add("System.ComponentModel.DataAnnotations");
    Settings.AdditionalNamespaces.Add("System.ComponentModel.DataAnnotations.Schema"); // Add this line
}
```

Modify your `<database>.tt` file and change:
```cs
column.Attributes.Add("[Key]");
```
to
```cs
column.Attributes.Add(string.Format("[Key, Column(Order = {0})]", column.Ordinal));
```

**Comment by sjh37** [comment]
@nick5454 This has now been released in v3.4.0
